pkg/srvconn/sftp_asset.go

```diff
@@ -552,7 +552,7 @@ func (ad *AssetDir) checkExpired() {
 
 func (ad *AssetDir) GetRealPath(sftpSess *SftpSession, path string) string {
 	realPath := filepath.Join(sftpSess.rootDirPath, strings.TrimPrefix(path, "/"))
-	if ad.isFromWebTerminal && path != "" {
+	if ad.isFromWebTerminal && path != "" && strings.HasPrefix(path, sftpSess.rootDirPath) {
 		return path
 	}
 	return realPath
```

Comment:
The provided code snippet isn't formatted to be analyzed like it would be for a regular document. It seems too fragmented and lacks context for meaningful analysis. Please ensure you have the complete code that I can analyze for discrepancies between two versions, potential issues, optimizations, or other relevant metrics so I could provide precise insights.